dali/pipeline/operators/geometric/bb_flip_test.cc

```diff
@@ -33,213 +38,143 @@ constexpr int kBbStructSize = 4;
 
-    InjectTestData<float, 0>(*test_data_, batch_data);
+  ASSERT_EQ(anticipated_output_roi.size(), output_roi.size())
+                        << "Inconsistent sizes (input vs output)";
+  for (size_t i = 0; i < output_roi.size(); i++) {
+    EXPECT_GT(kEpsilon, std::fabs(output_roi[i] - anticipated_output_roi[i]))
+                  << "Error exceeds allowed value";
   }
+}
 
+}  // namespace
 
-  const OpSpec GetOperatorSpec(bool ltrb_coordinates_type,
-                               bool vertical, bool horizontal) noexcept {
-    flip_type_vertical_ = vertical;
-    flip_type_horizontal_ = horizontal;
-    return OpSpec("BbFlip")
-            .AddArg("ltrb", ltrb_coordinates_type)
-            .AddArg("vertical", vertical ? 1 : 0)
-            .AddArg("horizontal", horizontal ? 1 : 0)
-            .AddInput("bb_input", "cpu")
-            .AddOutput("bb_output", "cpu");
+class BbFlipTest : public testing::DaliOperatorTest {
+  GraphDescr GenerateOperatorGraph() const noexcept override {
+    GraphDescr graph("BbFlip");
+    return graph;
   }
 
 
- private:
-  const TestData *test_data_ = nullptr;
-  std::vector<std::vector<long int>> new_batch_size_;  //NOLINT
-  bool flip_type_vertical_, flip_type_horizontal_;
+ public:
+  BbFlipTest() : DaliOperatorTest(1, 1) {}
 };
 
-// XXX: `DALISingleOpTest` assumes, that input to the operator
-//      is always image and is templated by ImageType.
-//      Therefore this test had to be TYPED, regardless
-//      of the fact, that it's unnecessary.
-typedef ::testing::Types<RGB, BGR, Gray> Types;
-TYPED_TEST_CASE(BbFlipTest, Types);
-
-TYPED_TEST(BbFlipTest, VerticalWHTest) {
-  TensorList<CPUBackend> bb_test_data;
-  this->LoadBbData(bb_test_data, &wh_rois);
-  this->SetExternalInputs({std::make_pair("bb_input", &bb_test_data)});
-  this->RunOperator(this->GetOperatorSpec(false, true, false), .001);
-}
-
-
-TYPED_TEST(BbFlipTest, Vertical2PTest) {
-  TensorList<CPUBackend> bb_test_data;
-  this->LoadBbData(bb_test_data, &two_pt_rois);
-  this->SetExternalInputs({std::make_pair("bb_input", &bb_test_data)});
-  this->RunOperator(this->GetOperatorSpec(true, true, false), .001);
-}
-
+std::vector<Arguments> arguments = {
+        {{"horizontal", 1}, {"vertical", 0}},
+        {{"horizontal", 0}, {"vertical", 1}},
+        {{"horizontal", 0}, {"vertical", 0}},
+};
 
-TYPED_TEST(BbFlipTest, HorizontalWHTest) {
-  TensorList<CPUBackend> bb_test_data;
-  this->LoadBbData(bb_test_data, &wh_rois);
-  this->SetExternalInputs({std::make_pair("bb_input", &bb_test_data)});
-  this->RunOperator(this->GetOperatorSpec(false, false, true), .001);
+TEST_P(BbFlipTest, WhRoisTest) {
+  constexpr bool ltrb = false;
+  auto args = GetParam();
+  args.emplace("ltrb", ltrb);
+  for (auto test_sample : rois_wh) {
+    auto tlin = ToTensorList<CPUBackend>(test_sample[0]);
+    TensorListWrapper tlout;
+    this->RunTest<CPUBackend>(tlin.get(), tlout, args, testing::Verify<ltrb>);
```

Comment:
No, no, no. I won't have it! You have broken the tests. Instead of running the operator on a tensor list, you feed it with N single-element tensors. This is different from the original, doesn't follow our pipeline usage and makes it impossible to test tensor arguments (since you'll always have batch size = 1).

Reply:
Done